frontend/tests/controllers/ContestUpdateTest.php

```diff
@@ -123,6 +153,96 @@ public function testUpdateDefaultShowAllContestantsOption() {
         );
     }
 
+    /**
+     * Update when there are too many problems in the contest.
+     */
+    public function testUpdateWhenTooManyProblemsInContest() {
+        $contestData = \OmegaUp\Test\Factories\Contest::createContest();
+        $login = self::login($contestData['director']);
+
+        for ($i = 0; $i < MAX_PROBLEMS_IN_CONTEST - 1; $i++) {
+            $problemData = \OmegaUp\Test\Factories\Problem::createProblemWithAuthor(
+                $contestData['director'],
+                $login
+            );
+
+            $r = new \OmegaUp\Request([
+                'auth_token' => $login->auth_token,
+                'contest_alias' => $contestData['contest']->alias,
+                'problem_alias' => $problemData['request']['problem_alias'],
+                'points' => 100,
+                'order_in_contest' => $i + 1,
+            ]);
```

Comment:
We have been trying to avoid the use of the $r variable in the tests. I understand that there are still many places where it is being used, but eventually, what we want is to pass the request inline. This way, we ensure that we are explicitly passing the necessary parameters:
```suggestion
            $response = \OmegaUp\Controllers\Contest::apiAddProblem(
                new \OmegaUp\Request([
                    'auth_token' => $login->auth_token,
                    'contest_alias' => $contestData['contest']->alias,
                    'problem_alias' => $problemData['request']['problem_alias'],
                    'points' => 100,
                    'order_in_contest' => $i + 1,
                ])
            );
```